test/parallel/test-async-wrap-asyncresource-constructor.js

```diff
@@ -0,0 +1,29 @@
+'use strict';
+require('../common');
+
+// This tests that AsyncResource throws an error if bad parameters are passed
+
+const assert = require('assert');
+const AsyncResource = require('async_hooks').AsyncResource;
+const async_wrap = process.binding('async_wrap');
+
+/* without this, there is nothing to do,
+* and the constructor would return immediately
+*/
+async_wrap.async_hook_fields[async_wrap.constants.kInit] = 1;
```

Comment:
updated - also, I am happy to submit a PR for the fix as well. let me know if you'd like that